app/request_handlers/handlers/login.js

```diff
@@ -0,0 +1,83 @@
+const { pool } = require('../../database/pool.js');
+const { KNOWN_USER_COOKIE_DURATION_MS } = require("../../constants.js");
+
+const argon2 = require('argon2');
+const crypto = require("crypto");
+
+const tokenStorage = {};
```

Comment:
Also, `tokenStorage` probably doesn't belong in `login.js` in the long-term, because anonymous users will need to create and store tokens as well. Their usernames would be some default value like "guest".